sdk/servicebus/service-bus/src/receivers/shared.ts

```diff
@@ -202,14 +202,29 @@ function settleMessage(
   entityPath: string,
   options?: DispositionStatusOptions
 ): Promise<void> {
-  if (!message.delivery) {
-    throw new Error("A peeked message cannot be settled.");
+  const isDeferredMessage = !message.delivery.link;
```

Comment:
Can `message.delivery` ever be undefined, or was the check that has now been removed never truthy?

Reply:
It would be undefined on `ServiceBusReceviedMessage`, but not on `ServiceBusMessageImpl`. I have another method `throwErrorIfInvalidOperationOnMessage()` which we call before trying to settle the message which has the checks that you see removed here